.github/actions/maven-build/action.yml

```diff
@@ -0,0 +1,280 @@
+  skip-tests-label:
+    description: The label name for skipping tests
+    required: false
+    default: 'skip-tests'
+  jacoco-report-name:
+    description: Jacoco report name. It will be used determine the path of the execution data file generated by `jacoco:merge` and consumed by `jacoco:report`
+    required: false
+  jacoco-report-options:
+    description: Jacoco report additional options
+    required: false
+  target-folder-upload-name:
+    description: The name of the artifact to upload the target folders
+    required: false
+  m2-cache-exclusion-pattern:
+    description: The exclusion pattern to be applied while creating cache for the Maven m2 repository. By default `org/activiti` and `com/alfresco` are excluded
+    required: false
+  m2-current-build-upload-name:
+    description: The name of the artifact to upload m2 repository related to artifacts produced by the current build
+    required: false
+  uploaded-artifacts-retention-days:
+    description: The number of days to retain the uploaded artifacts
+    required: false
+    default: "7"
+
+outputs:
+  m2-uploaded-group-path:
+    description: The path to the group id uploaded from m2 repository
+    value: ${{ steps.m2-inclusion-pattern.outputs.group-path }}
+  skip-tests:
+    description: Whether the tests were skipped during build or not.
+    value: ${{ steps.maven-configure.outputs.skip-tests }}
+
+
+runs:
+  using: composite
+  steps:
+    - uses: Alfresco/alfresco-build-tools/.github/actions/maven-configure@e113e1940de607a5d1ecbb3ff5d2973618cbb630
+      id: maven-configure
+      with:
+        java-version: ${{ inputs.java-version }}
+        java-distribution: ${{ inputs.java-distribution }}
+        maven-version: ${{ inputs.maven-version }}
+        maven-build-cache-key: ${{ inputs.maven-build-cache-key }}
+        maven-build-cache-skip-label: ${{ inputs.maven-build-cache-skip-label }}
+        skip-tests-label: ${{ inputs.skip-tests-label }}
+        verbose: ${{ inputs.verbose }}
+        preview-label: ${{ inputs.preview-label }}
+
+    - name: Update pom files to the preview version
+      if: steps.maven-configure.outputs.preview-version != ''
+      uses: Alfresco/alfresco-build-tools/.github/actions/maven-update-pom-version@78bff56d1f8dbeb0ccfa6defbd1d68a040034f05
+      env:
+        MAVEN_USERNAME: ${{ inputs.maven-username }}
+        MAVEN_PASSWORD: ${{ inputs.maven-password }}
+      with:
+        version: ${{ steps.maven-configure.outputs.preview-version }}
+        maven-cli-opts: ${{ steps.maven-configure.outputs.maven-options }} ${{ inputs.maven-resolver-transport-options }}
+        property-to-update: ${{ inputs.property-to-update }}
+
+    - name: Get project information
+      id: project-info
+      shell: bash
+      run: |
+        VERSION=$(yq e '.project.version' pom.xml)
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+    - name: Set version env variable
+      if: github.event_name == 'push' || env.IS_PREVIEW == 'true'
```

Comment:
why do we set this env variable only on push or preview? why do we need it in env?